src/test/groovy/com/mechanitis/demo/spock/OrderingServiceTest.groovy

```diff
@@ -0,0 +1,100 @@
+package com.mechanitis.demo.spock
+
+import com.synacy.gradprogram.spock.exercise.*
+import spock.lang.Specification
+
+class OrderingServiceTest extends Specification {
+
+    OrderingService service
+    Cart cart
+
+    def items = [
+            new Item("Chicken Cordon", 25, ItemType.FOOD),
+            new Item("Trouser", 1500, ItemType.CLOTHING),
+            new Item("Smart-TV", 3500, ItemType.APPLIANCE),
+            new Item("IPhone14Pro", 62000, ItemType.GADGET),
+            new Item("Pancit Malabon", 110, ItemType.FOOD)
+    ]
+
+    def setup() {
+        service = new OrderingService()
+    }
+
+    def "Validate the Items if there have food Order on the Cart"() {
+        when:
+        boolean validateCartItem = service.cartContainsFoodItem(cart = new Cart(UUID.randomUUID(), items))
+
+        then:
+        validateCartItem
+    }
+
+    def "Verify the total amount of cost items in the cart"() {
+        when:
+        double totalAmountPrice = service.calculateTotalCostOfCart(cart = new Cart(UUID.randomUUID(), items))
+        then:
+        totalAmountPrice
+    }
+
+    def "Check if the cart items eligible for discount"() {
+        given:
+        cart = new Cart(UUID.randomUUID(), items)
+
+        when:
+        boolean eligibleDiscountItem = service.isCartEligibleForDiscount(cart)
+
+        then:
+        eligibleDiscountItem
+    }
+
+    def "Check item if eligible to apply the discount"() {
+        given:
+        cart = new Cart(UUID.randomUUID(), items)
+        List totalDiscount = cart.getItems().collect() { it.getCost() }
```

Comment:
Let's try to make things as simple as we can. Let's try to avoid anything that does any kind of looping. My suggestion would be to use fewer items to make it easier.